### PR TITLE
Add cdci artifact registry

### PR DESCRIPTION
**Pull request by jonfoong**
This PR:

- allows CDCI by building and pushing to artifact registry on git push
- inherits gcp token from github secrets and passes it to docker during build

### CODE REVIEW

File: .github/workflows/docker-build-and-push.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      uses: docker/setup-buildx-action@v2
+
+    - name: Log in to Google Cloud
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+
+    - name: Get OAuth2 access token and login to Docker
+      run: |
+        ACCESS_TOKEN=$(gcloud auth print-access-token)
```
Comment (jeremychia) [inline .github/workflows/docker-build-and-push.yaml:26]: we just need to make sure that it doesn't print anything out into the command line

**Review by jeremychia — APPROVED** [review]

File: .github/workflows/docker-build-and-push.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - add-CDCI-artifact-registry  # Change this to your target branch
```
Comment (jeremychia) [inline .github/workflows/docker-build-and-push.yaml]: should this be main?
```suggestion
on:
  push:
    branches:
      - main
```

Comment (jonfoong) [reply .github/workflows/docker-build-and-push.yaml]: yes you're right this was for testing on branch

File: .github/workflows/docker-build-and-push.yaml (hunk 1 above)
Comment (jonfoong) [reply .github/workflows/docker-build-and-push.yaml:26]: good call, I checked and looks clear

**Pull request merged**